app/services/address_validation_service.rb

```diff
@@ -30,25 +30,38 @@ def po_box?(validation)
     #   does not normalize street1 so check case insensitive with optional periods and
     #   space, e.g. look for 'po box', 'P. O. Box'
     # https://www.easypost.com/errors-guide
+    false unless validation.present?
+
     validation.street1.match(/P\.?\s*O\.?\s*BOX/i) || (
       !validation.verifications.delivery.success &&
       validation.verifications.delivery.errors[0].code == 'E.BOX_NUMBER.INVALID'
     )
   end
 
+  def duplicate_unit?(validation)
+    # Check for duplicate apt/unit no on street lines 1 and 2
+    validation.present? && !validation.street2.empty? &&
+      validation.street1.end_with?(validation.street2)
+  end
+
   def invalid?
     # we don't treat timeouts as "invalid" since we want to allow them to proceed
     Rails.logger.warn('Address validation: Easypost request timed out') if timeout?
 
     # we do not accept PO Boxes
-    if @validation.present? && po_box?(@validation)
+    if po_box?(@validation)
```

Comment:
moved this into the `po_box` function because rubocop said this method was getting to complex